buzzad-benefit/app/src/main/java/com/buzzvil/buzzad/benefit/sample/publisher/carousel/NativeCarouselActivity.java

```diff
@@ -0,0 +1,192 @@
+
+        @Override
+        public void onAdsLoaded(final Collection<NativeAd> nativeAds) {
+            if (nativeAds.isEmpty()) {
+                onEmpty();
+            } else {
+                NativeCarouselActivity.this.onAdsLoaded(nativeAds);
+            }
+        }
+    };
+
+    private final NativeAdEventListener nativeAdEventListener = new NativeAdEventListener() {
+        @Override
+        public void onImpressed(final NativeAd nativeAd) {
+        }
+
+        @Override
+        public void onClicked(final NativeAd nativeAd) {
+        }
+
+        @Override
+        public void onRewardRequested(final NativeAd nativeAd) {
+        }
+
+        @Override
+        public void onRewarded(
+                final NativeAd nativeAd,
+                final RewardResult nativeAdRewardResult
+        ) {
+        }
+
+        @Override
+        public void onParticipated(final NativeAd nativeAd) {
+        }
+    };
+
+
+    @Override
+    public void onCreate(final Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        binding = ActivityNativeCarouselBinding.inflate(getLayoutInflater());
+        setContentView(binding.getRoot());
+        adapter = new CarouselAdapter(infiniteLoop);
+        initCarousel();
+        loadAds();
+    }
+
+    private void initCarousel() {
+        binding.carousel.setAdapter(adapter);
+
+        // ViewPager와 비슷하게, 스크롤이 중간에서 멈추면 한쪽 아이템을 선택하여 다 보일 때까지 스크롤
+        new PagerSnapHelper().attachToRecyclerView(binding.carousel);
+
+        binding.carousel.setLayoutManager(layoutManager);
+        // 아이템 간에 여백을 넣는 Decoration 추가
+        binding.carousel.addItemDecoration(new PaddingDividerDecoration(itemPaddingDp));
+    }
+
+    private void loadAds() {
+        new NativeAdLoader(unitId).loadAds(
+                adsLoadedListener,
+                new NativeAdLoaderParams.Builder()
+                        .count(5)
+                        .build()
+        );
+    }
+
+    private void onError() {
+        // 에러가 발생하면 지면을 표시하지 않는다.
+        binding.placement.setVisibility(View.GONE);
+        Log.d(TAG, "An error occurred while loading ads");
+    }
+
+    private void onEmpty() {
+        // 보여줄 광고가 없으면 지면을 표시하지 않는다.
+        binding.placement.setVisibility(View.GONE);
+        Log.d(TAG, "The ad list is empty");
+    }
+
+    private void onAdsLoaded(final Collection<NativeAd> nativeAds) {
+        binding.placement.setVisibility(View.VISIBLE);
+        for (final NativeAd nativeAd : nativeAds) {
+            nativeAd.addNativeAdEventListener(nativeAdEventListener);
+        }
+
+        final List<CarouselItem> items = buildCarouselItems(nativeAds);
+        adapter.submitList(items);
+        if (infiniteLoop) {
+            final RecyclerView.LayoutManager layoutManager = binding.carousel.getLayoutManager();
+            if (layoutManager != null) {
+                // 첫번째 광고 앞에 마지막 아이템의 부분적으로 노출하기
+                layoutManager.scrollToPosition(items.size() * 10000);
```

Comment:
이 부분 주석이랑 `* 10000`의 의미가 잘 와닿지 않는데, 만약 *2를 하면 뒤로는 무한루프(처럼 보이는)이지만, 앞으로는 1세트밖에 없어서 무한루프처럼 안보이니까 10000이라는 적당히 큰 수를 곱해서 앞뒤로 무한루프가 되도록 하신게 맞나요?

위 내용이 맞다면 `layoutManager.scrollToPosition(items.size() * (Integer.MAX_VALUE / 2));` 이런식으로 작성하는 것도 어댑터 코드와 연관성이 있어서 이해하기 쉬울 것 같습니다.

```suggestion
                // 스크롤이 무한 루프처럼 보이도록 중간에서 시작
                layoutManager.scrollToPosition(items.size() * (Integer.MAX_VALUE / 2));
```

Reply:
스크롤이 무한 루프처럼 보이도록 중간에서 시작 -> 스크롤이 '앞뒤로' 무한 루프처럼 보이도록 중간에서 시작

Reply:
좋습니다.

Reply:
`adapter.getItemCount() / 2`를 사용하는 것도 가능하다면 나쁘지 않을 것 같네요.

Reply:
@MiJey items.size() * (Integer.MAX_VALUE / 2) 이거, 아이템 크기가 2보다 크면 MAX_VALUE를 넘어갈 것 같습니다.

Reply:
엇 그러게요... 그렇담 `items.size() * (Integer.MAX_VALUE / items.size() / 2)`... (지저분)

Reply:
* 10000 괜찮은 것 같습니다

Reply:
오히려 설득당해버리셨군요ㅋㅋ 주석만 좀 변경해보겠습니다.

Reply:
https://github.com/Buzzvil/buzz-sdk-samples/pull/74/commits/ae8f7ae879050d1a106865cb67902f3326d84407 이렇게 수정하였습니다.